scripts/upgrade.ts

```diff
@@ -1,54 +1,228 @@
-import fs from "fs";
-import path from "path";
 import { StateDeployHelper } from "../helpers/StateDeployHelper";
-import { BigNumber } from "ethers";
+import { ethers } from "hardhat";
+import { StateTestContractMigrationSteps } from "../helpers/MigrationHelper";
+import { Contract } from "ethers";
+import fs from "fs";
 
-const pathOutputJson = path.join(__dirname, "./upgrade_output.json");
+/*
+1. deploy stateV2 to mumbai from feature/state-v3 branch
+2. run transit-state script
+3. cp .openzeppelin/* ../../contracts/.openzeppelin/
+4. update addreess and block number in data
+5. run this script
+*/
 
-async function main() {
-  const stateDeployHelper = await StateDeployHelper.initialize(null, true);
-  const proxyAddress = "0xa513E6E4b8f2a923D98304ec87F64353C4D5C853";
-
-  const { state, verifier, smtLib, poseidon1, poseidon2, poseidon3 } =
-    await stateDeployHelper.upgradeToStateV2_migration(proxyAddress);
-
-  const outputJson = {
-    state: state.address,
-    verifier: verifier.address,
-    smtLib: smtLib.address,
-    poseidon1: poseidon1.address,
-    poseidon2: poseidon2.address,
-    poseidon3: poseidon3.address,
-    network: process.env.HARDHAT_NETWORK,
+export async function getDataFromContract(
+  contract: Contract,
+  id: bigint,
+  state: bigint,
+  fileName: string
+): Promise<any> {
+  const stateInfoHistoryLengthById = await contract.getStateInfoHistoryLengthById(id);
+  const latestStateInfoById = await contract.getStateInfoById(id);
+  const stateInfoByIdAndState = await contract.getStateInfoByIdAndState(id, state);
+  const stateInfoHistory = await contract.getStateInfoHistoryById(id, 0, 3);
+  const root = await contract.getGISTRoot();
+  const rootInfo = await contract.getGISTRootInfo(root);
+  const gistProof = await contract.getGISTProof(id);
+  const gistProofByRoot = await contract.getGISTProofByRoot(id, root);
+
+  const result = {
+    stateInfoHistoryLengthById,
+    latestStateInfoById,
+    stateInfoByIdAndState,
+    stateInfoHistory,
+    root,
+    gistProof,
+    rootInfo,
+    gistProofByRoot,
   };
-  fs.writeFileSync(pathOutputJson, JSON.stringify(outputJson, null, 1));
 
-  const id = BigNumber.from("0x000c91060cf3aa883f1d50203499abcb06a0eedcf1971e15afc88475712b1202");
+  fs.writeFileSync(fileName, JSON.stringify(result, null, 2));
```

Comment:
Generally, it's not the best approach to make a function do not related actions like writing to file here.